src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -111,10 +132,37 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
                 throw new ArgumentNullException(nameof(controllerContext));
             }
 
+            var valueProvider = await CompositeValueProvider.CreateAsync(
+                controllerContext,
+                controllerContext.ValueProviderFactories);
+
+            return await BindModelAsync(parameter, controllerContext, valueProvider);
+        }
+
+        public async Task<ModelBindingResult?> BindModelAsync(
+            ParameterDescriptor parameter,
+            ControllerContext controllerContext,
+            IValueProvider valueProvider)
```

Comment:
This method needs to take the value provider as a parameter to avoid creating it over and over